### PR TITLE
[Banner] Support Dynamic Type in Banner

### LINKED ISSUE #7621: [Banner] Internal issue: b/135474807
This was filed as an internal issue. If you are a Googler, please visit [b/135474807](http://b/135474807) for more details.

---
#### Internal data
- Associated internal bug: [b/135474807](http://b/135474807)

### PR DESCRIPTION
**Pull request by fluxminds**
Closes https://github.com/material-components/material-components-ios/issues/7621.

Screenshot:

|Adjusted Sizing Category 1 | Adjusted Sizing Category 2|
|---|---|
|IMG_0789|IMG_0788|

### CODE REVIEW

File: components/Banner/src/MDCBannerView.m [hunk 1]
```diff
@@ -505,4 +507,54 @@ - (BOOL)isAbleToFitTextLabel:(UILabel *)textLabel withWidthLimit:(CGFloat)widthL
...
+    textFont = [textFont mdc_scaledFontForTraitEnvironment:self];
+  }
+  self.textLabel.font = textFont;
+}
+
+- (void)updateButtonFont {
+  self.leadingButton.mdc_adjustsFontForContentSizeCategory =
+      self.mdc_adjustsFontForContentSizeCategory;
+  self.trailingButton.mdc_adjustsFontForContentSizeCategory =
+      self.mdc_adjustsFontForContentSizeCategory;
```
Comment (jverkoey) [inline components/Banner/src/MDCBannerView.m]: Can this method be folded in to the setter for mdc_adjustsFontForContentSizeCategory? I'd expect these values to be kept in sync there.

File: components/Banner/tests/unit/BannerViewTests.m [hunk 2]
```diff
@@ -0,0 +1,100 @@
+// Copyright 2017-present the Material Components for iOS authors. All Rights Reserved.
```
Comment (jverkoey) [inline components/Banner/tests/unit/BannerViewTests.m]: Please update the copyright date.

File: components/Banner/tests/unit/BannerViewTests.m [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    // Then
+    CGFloat actualTextFontSize = bannerView.textLabel.font.pointSize;
+    XCTAssertGreaterThan(actualTextFontSize, originalTextFontSize);
+    CGFloat actualButtonFontSize =
+        [bannerView.leadingButton titleFontForState:UIControlStateNormal].pointSize;
+    XCTAssertGreaterThan(actualButtonFontSize, originalButtonFontSize);
+  }
+}
+
+@end
```
Comment (jverkoey) [inline components/Banner/tests/unit/BannerViewTests.m:100]: Are we able to add snapshot tests for this behavior? They may be preferable to logic tests in this case.

File: components/Banner/tests/unit/BannerViewTests.m (hunk 2 above)
Comment (fluxminds) [reply components/Banner/tests/unit/BannerViewTests.m]: Addressed in f48539c3d7cd07ad36955c1a13b11c89ebecaaa4.

File: components/Banner/src/MDCBannerView.m (hunk 1 above)
Comment (fluxminds) [reply components/Banner/src/MDCBannerView.m]: Addressed in f48539c3d7cd07ad36955c1a13b11c89ebecaaa4.

File: components/Banner/tests/unit/BannerViewTests.m (hunk 3 above)
Comment (fluxminds) [reply components/Banner/tests/unit/BannerViewTests.m:100]: Added in f428e8e7b6110f0c28bb638c8a9317983a74048d

File: components/Banner/tests/snapshot/MDCBannerSnapshotTests.m [hunk 4]
```diff
@@ -37,7 +76,7 @@ - (void)setUp {
 
   // Uncomment below to recreate all the goldens (or add the following line to the specific
   // test you wish to recreate the golden for).
-  //  self.recordMode = YES;
+  self.recordMode = YES;
```
Comment (jverkoey) [inline components/Banner/tests/snapshot/MDCBannerSnapshotTests.m]: Don't forget to comment back out.

**Review by jverkoey — APPROVED** [review]

File: components/Banner/tests/snapshot/MDCBannerSnapshotTests.m (hunk 4 above)
Comment (fluxminds) [reply components/Banner/tests/snapshot/MDCBannerSnapshotTests.m]: Ah...Thanks for catching it!! I forgot it. Addressed in d93481634003f2329f773256e9aef6ce9df9e3ae.

**Pull request merged**